device/cuda/src/seeding/seed_finding.cu

```diff
@@ -162,28 +200,22 @@ vecmem::data::vector_buffer<seed> seed_finding::operator()(
     const sp_grid_const_view& g2_view,
     const std::vector<unsigned int>& grid_sizes) const {
 
-    // Get the prefix sum for the spacepoint grid using buffer.
-    const device::prefix_sum_t sp_grid_prefix_sum = device::get_prefix_sum(
-        grid_sizes, (m_mr.host ? *(m_mr.host) : m_mr.main));
+    // Create prefix sum buffer and its view
+    vecmem::data::vector_buffer sp_grid_prefix_sum_buff =
+        make_prefix_sum_buff(grid_sizes, m_copy, m_mr);
+    vecmem::data::vector_view<device::prefix_sum_element_t>
+        sp_grid_prefix_sum_view = sp_grid_prefix_sum_buff;
```

Comment:
Same here. You can just use the buffer directly in CUDA code, no need to explicitly cast it to a view.